src/AppBundle/Controller/AdminStatsController.php

```diff
@@ -1,103 +1,279 @@
+        $endDb = date('Ymd000000', strtotime($end));
 
-            'url' => $url,
-            'project' => $project,
-            'wikiName' => $wikiName,
+        // TODO: Fix this - inactive admins aren't getting shown
+        $query = "
+            SELECT user_name, user_id
+            ,SUM(IF( (log_type='delete'  AND log_action != 'restore'),1,0)) AS mdelete
+            ,SUM(IF( (log_type='delete'  AND log_action  = 'restore'),1,0)) AS mrestore
+            ,SUM(IF( (log_type='block'   AND log_action != 'unblock'),1,0)) AS mblock
+            ,SUM(IF( (log_type='block'   AND log_action  = 'unblock'),1,0)) AS munblock
+            ,SUM(IF( (log_type='protect' AND log_action !='unprotect'),1,0)) AS mprotect
+            ,SUM(IF( (log_type='protect' AND log_action  ='unprotect'),1,0)) AS munprotect
+            ,SUM(IF( log_type='rights',1,0)) AS mrights
+            ,SUM(IF( log_type='import',1,0)) AS mimport
+            ,SUM(IF(log_type !='',1,0)) AS mtotal
+            FROM $loggingTable
+            JOIN $userTable ON user_id = log_user
+            WHERE  log_timestamp > '$startDb' AND log_timestamp <= '$endDb'
+              AND log_type IS NOT NULL
+              AND log_action IS NOT NULL
+              AND log_type IN ('block', 'delete', 'protect', 'import', 'rights')
+            GROUP BY user_name
+            HAVING mdelete > 0 OR user_id IN ($adminIds)
+            ORDER BY mtotal DESC";
 
-            'users' => $users,
-        ]);
+        $res = $conn->prepare($query);
+        $res->execute();
+
+        // Fetch all the information out.  Because of pre-processing done
+        // in the query, we can use this practically raw.
+        $users = $res->fetchAll();
+
+        // Get the total number of admins, the number of admins without
+        // action, and then run percentage calculations on the same.
+        $adminCount = sizeof($adminIdArr);
+
+        foreach ($users as $row) {
+            if ($row["mtotal"] == 0) {
+                $adminsWithoutAction++;
+            }
+        }
+
+        if ($adminCount > 0) {
+            $adminsWithoutActionPct = $adminsWithoutAction/$adminCount;
+        }
+
+        // Pull the admins from the API, for merging.
+        $data = $api->getAdmins($project);
+
+        // Combine the two arrays.  We can't use array_merge here because
+        // the arrays contain fundamentally different data.  Instead, it's
+        // done by hand.  Only two values are needed, edit count and groups.
+        foreach ($users as $key => $value) {
+            // FIXME: workaround to get the page just to show some data,
+            //  since the API only returns 500 results and the admin returned
+            //  by the query may not have been returned by the API
+            if (empty($data[$value["user_name"]])) {
+                $data[$value["user_name"]] = [
+                    'editcount' => 0,
+                    'groups' => '',
+                ];
+            }
```

Comment:
We need to either use a bot account or use this `massApi` function https://github.com/x-tools/xtools-rebirth/blob/master/src/AppBundle/Helper/ApiHelper.php#L341-L357

For us I think a bot account is preferred, but not sure how we could set that up in such a way to not conflict with the way 3rd wikis are set up. Maybe yet another set of config values in parameters.yml? :)